flake.nix

```diff
@@ -21,73 +21,111 @@
-                  "book"
-                  "contrib"
-                  "docs"
-                  "README.md"
-                  "shell.nix"
-                  "default.nix"
-                  "grammars.nix"
-                  "flake.nix"
-                  "flake.lock"
-                ];
-            };
+      pkgConfig = common: {
+        helix-term.overrides.fix-build.overrideAttrs = prev: {
+          src = filteredSource;
 
-            # disable fetching and building of tree-sitter grammars in the helix-term build.rs
-            HELIX_DISABLE_AUTO_GRAMMAR_BUILD = "1";
+          # disable fetching and building of tree-sitter grammars in the helix-term build.rs
+          HELIX_DISABLE_AUTO_GRAMMAR_BUILD = "1";
 
-            buildInputs = (prev.buildInputs or []) ++ [common.cCompiler.cc.lib];
+          buildInputs = ncl.addBuildInputs prev [common.config.cCompiler.package.cc.lib];
 
-            # link languages and theme toml files since helix-term expects them (for tests)
-            preConfigure = ''
-              ${prev.preConfigure or ""}
-              ${
-                lib.concatMapStringsSep
-                "\n"
-                (path: "ln -sf ${mkRootPath path} ..")
-                ["languages.toml" "theme.toml" "base16_theme.toml"]
-              }
-            '';
+          # link languages and theme toml files since helix-term expects them (for tests)
+          preConfigure = ''
+            ${prev.preConfigure or ""}
+            ${
+              lib.concatMapStringsSep
+              "\n"
+              (path: "ln -sf ${mkRootPath path} ..")
+              ["languages.toml" "theme.toml" "base16_theme.toml"]
+            }
+          '';
 
-            meta.mainProgram = "hx";
-          };
+          meta.mainProgram = "hx";
         };
-        shell = common: prev: {
+        # Wrap helix with runtime
+        helix-term.wrapper = _: old: let
+          inherit (common) pkgs;
+          makeOverridableHelix = old: config: let
+            grammars = pkgs.callPackage ./grammars.nix config;
+            runtimeDir = pkgs.runCommand "helix-runtime" {} ''
+              mkdir -p $out
+              ln -s ${mkRootPath "runtime"}/* $out
+              rm -r $out/grammars
+              ln -s ${grammars} $out/grammars
+            '';
+            helix-wrapped =
+              common.internal.pkgsSet.utils.wrapDerivation old
+              {
+                nativeBuildInputs = [pkgs.makeWrapper];
+                makeWrapperArgs = config.makeWrapperArgs or [];
+              }
+              ''
+                rm -rf $out/bin
+                mkdir -p $out/bin
+                ln -sf ${old}/bin/* $out/bin/
+                wrapProgram "$out/bin/hx" ''${makeWrapperArgs[@]} --set HELIX_RUNTIME "${runtimeDir}"
+              '';
+          in
+            helix-wrapped
+            // {override = makeOverridableHelix old;};
+        in
+          makeOverridableHelix old {};
+        helix-term.shell = prev: {
```

Comment:
Based on the docs I think this should move up to `config`, right? Without that I don't have rust-analyzer in my `nix develop` shell

Reply:
Yes, that would be correct, I forgot to test the devshell...

Reply:
Updated the devshell, should be working now